pkg/clients/aws/route53.go

```diff
@@ -73,107 +72,48 @@ func (c *awsClient) GetDNSName() string {
 	return "Route53"
 }
 
-func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken string, domain string, cr *certmanv1alpha1.CertificateRequest) (fqdn string, err error) {
+func (c *awsClient) AnswerDNSChallenge(reqLogger logr.Logger, acmeChallengeToken string, domain string, cr *certmanv1alpha1.CertificateRequest, dnsZone string) (fqdn string, err error) {
 	fqdn = fmt.Sprintf("%s.%s", cTypes.AcmeChallengeSubDomain, domain)
 	reqLogger.Info(fmt.Sprintf("fqdn acme challenge domain is %v", fqdn))
 
-	if fedramp {
-		zone, err := c.client.GetHostedZone(&route53.GetHostedZoneInput{Id: &fedrampHostedZoneID})
-		if err != nil {
-			reqLogger.Error(err, err.Error())
-			return "", err
-		}
-
-		input := &route53.ChangeResourceRecordSetsInput{
-			ChangeBatch: &route53.ChangeBatch{
-				Changes: []*route53.Change{
-					{
-						Action: aws.String(route53.ChangeActionUpsert),
-						ResourceRecordSet: &route53.ResourceRecordSet{
-							Name: &fqdn,
-							ResourceRecords: []*route53.ResourceRecord{
-								{
-									Value: aws.String(fmt.Sprintf("\"%s\"", acmeChallengeToken)),
-								},
+	input := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: []*route53.Change{
+				{
+					Action: aws.String(route53.ChangeActionUpsert),
+					ResourceRecordSet: &route53.ResourceRecordSet{
+						Name: &fqdn,
+						ResourceRecords: []*route53.ResourceRecord{
+							{
+								Value: aws.String(fmt.Sprintf("\"%s\"", acmeChallengeToken)),
 							},
-							TTL:  aws.Int64(resourceRecordTTL),
-							Type: aws.String(route53.RRTypeTxt),
 						},
+						TTL:  aws.Int64(resourceRecordTTL),
+						Type: aws.String(route53.RRTypeTxt),
 					},
 				},
-				Comment: aws.String(""),
 			},
-			HostedZoneId: zone.HostedZone.Id,
-		}
-
-		reqLogger.Info(fmt.Sprintf("updating hosted zone %v", zone.HostedZone.Name))
+			Comment: aws.String(""),
+		},
+	}
 
-		result, err := c.client.ChangeResourceRecordSets(input)
+	if fedramp {
+		zone, err := c.client.GetHostedZone(&route53.GetHostedZoneInput{Id: &fedrampHostedZoneID})
 		if err != nil {
-			reqLogger.Error(err, result.GoString(), "fqdn", fqdn)
+			reqLogger.Error(err, err.Error())
 			return "", err
 		}
-
-		return fqdn, nil
+		input.HostedZoneId = zone.HostedZone.Id
+	} else {
```

Comment:
instead of the else here you can set input.HostedZoneId = &dnsZone and overwrite it if it is fedramp